### DISCUSSION TITLE
airflow in docker : the airflow default user

### DISCUSSION THREAD

**Discussion by zyt1024** [apache/airflow#27739] [Q&A]
I have deployed airflow using `docker-compose.yml` provided in the tutorial，when I user `docker exec -it  airflow_weorker bash` enter the docker,I found the user name is default. I know I could user `docker exec -it  -u xxx airflow_weorker bash` to select root to enter the docker. However, the worker runtime uses default by default, which is suitable for me to modify？
![image](https://user-images.githubusercontent.com/42999008/202421153-569ecbcb-07b8-460c-bea2-2de3154af6ed.png)

**Comment by SChalupaBraiins** [comment]
I have a similiar question, which user and how is used in container by default?
My case - if I set 
USER airflow
in Dockerfile, user airflow is used indeed, default context is GID=0(root), but I am not able to add any permission group to enhance permission, even if I set it all in Dockerfile.
```shell
airflow@dbcce1941e7b:/opt/airflow$ id
uid=1002(airflow) gid=0(root) groups=0(root)
airflow@dbcce1941e7b:/opt/airflow$ id airflow
uid=1002(airflow) gid=0(root) groups=0(root),1002(airflow),998(docker)
```

**Comment by potiuk** [comment]
Answering the both of you: 

Airflow Image is redshift compatible which means that it can run with any user as long as GID is 0 (so GId=0 is absolutely necessary and you can't change it).  All that providing that you use the image entrypoint and not `bash` as your entrypoint (which is `/entrypoint`)

You can find all information about it (including the way how you can exec and run custom commands including starting `bash`) in the documentation https://airflow.apache.org/docs/docker-stack/entrypoint.html

**Comment by SChalupaBraiins** [comment]
> I have a similiar question, which user and how is used in container by default? My case - if I set USER airflow in Dockerfile, user airflow is used indeed, default context is GID=0(root), but I am not able to add any permission group to enhance permission, even if I set it all in Dockerfile.
> 
> ```shell
> airflow@dbcce1941e7b:/opt/airflow$ id
> uid=1002(airflow) gid=0(root) groups=0(root)
> airflow@dbcce1941e7b:/opt/airflow$ id airflow
> uid=1002(airflow) gid=0(root) groups=0(root),1002(airflow),998(docker)
> ```

Thank you @potiuk for your comment. I've already found a way for my issue.
GID=0 is necessary indeed. I set up an **airflow** system user and **docker** group on host envrionment and the container, both with the same UID and GID.

The reason of my problem was this line in docker_compose.yml:
`user: "${AIRFLOW_UID:-50000}:0"`
The default GID:0 causes a different user context for default user. When you avoid it, everything works fine
`user: "${AIRFLOW_UID:-50000}"`

I use this in Dockerfile, if anyone interested (use your own UID, GID numbers from the docker host).
The docker group has access to docker.sock on host, allowing me to call docker from the airflow container, I use it for some dbt transformations scheduling.

```
# creating/updating the docker group if it doesn't exist
RUN if getent group docker ; then groupmod -g 998 docker; else groupadd -g 998 docker; fi
# updating the airflow user's UID and adding to the docker group
RUN usermod -u 1002 -g 0 -aG docker airflow
# switching to user airflow with default group root (GID 0)
USER airflow
```
